publisher/internal/analytics/ethereum/operations.go

```diff
@@ -257,13 +244,10 @@ func (add *Addition) Process(mint WrappedEventLog) error {
 	if !isUniswapPositionsNFT(mintLog.Data) {
 		return fmt.Errorf("not Uniswap Positions NFT.\n")
 	}
-	//TODO: Create a 'position initialization' function
-	addPosition := &Position{
-		Address:   mintLog.Address,
-		TxHash:    mintLog.TransactionHash,
-		LowerTick: int(convertHexToBigInt(mintLog.Topics[2]).Int64()),
-		UpperTick: int(convertHexToBigInt(mintLog.Topics[3]).Int64()),
-	}
+
+	addPosition := newPosition(mintLog)
+	addPosition.LowerTick = int(convertHexToBigInt(mintLog.Topics[2]).Int64())
+	addPosition.UpperTick = int(convertHexToBigInt(mintLog.Topics[3]).Int64())
 	add.Position = *addPosition
```

Comment:
After `newPosition` the result is immediately dereferenced. It would be much cleaner to return a struct instead of a pointer to a struct directly, so this could be possible:
```
  add.Position = newPosition(mintLog)
```